internal/cephfs/controllerserver.go

```diff
@@ -712,3 +712,26 @@ func (cs *ControllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
 
 	return &csi.DeleteSnapshotResponse{}, nil
 }
+
+// validateSnapshotRetention verifies if the referred subvolume has
+// 'snapshot-retention' feature.
+func validateSnapshotRetention(ctx context.Context, volID volumeID, cr *util.Credentials, volOptions *volumeOptions) (bool, error) {
+	info, err := getSubVolumeInfo(ctx, volOptions, cr, volID)
+	if err != nil {
+		// For older Ceph versions where 'features' was not returned in the info output,
+		// or `subvolume info` command is not supported,it is assumed that these do not
+		// support the intended features.
+		// The Purge operation should then continue without snapshot retention.
+		if errors.Is(err, ErrInvalidCommand) {
+			return false, nil
+		}
+		return false, status.Error(codes.Internal, err.Error())
+	}
+	// check if subvolume supports snapshot retention feature.
+	for _, feature := range info.Features {
+		if feature == "snapshot-retention" {
```

Comment:
why this feature will be set on the subvolume? during create or it will be auto-added by ceph when you create subvolume? please add a comment here

Reply:
Thanks. I will add a comment here.